command/agent/srv.go

```diff
@@ -0,0 +1,90 @@
+package agent
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"time"
+)
+
+const (
+	srvPollInterval  = 60 * time.Second
+	srvQuietInterval = 100 * time.Millisecond
+)
+
+// AgentSRV periodically polls an SRV record
+// And attempts to join the hosts supplied
+type AgentSRV struct {
+	agent      *Agent
+	srvrecords []string
+	logger     *log.Logger
+	replay     bool
+}
+
+// NewAgentSRV is used to create a new AgentSRV
+func NewAgentSRV(agent *Agent, logOutput io.Writer, replay bool, srvrecords []string) (*AgentSRV, error) {
+
+	// Initialize the AgentSRV
+	m := &AgentSRV{
+		agent:      agent,
+		srvrecords: srvrecords,
+		logger:     log.New(logOutput, "", log.LstdFlags),
+		replay:     replay,
+	}
+
+	// Start the background workers
+	go m.run()
+	return m, nil
+}
+
+// run is a long running goroutine that scans for new hosts periodically
+func (m *AgentSRV) run() {
+	hosts := make(chan *net.SRV)
+	poll := time.After(0)
+	var quiet <-chan time.Time
+	var join []string
+
+	for {
+		select {
+		case h := <-hosts:
+			// Format the host address
+			addr := fmt.Sprintf("%s:%d", h.Target, h.Port)
+
```

Comment:
Not if agentA is the only thing in the SRV record (it would be trying to join itself).

In our case, only a subset of the agents are in the SRV record, we call them 'anchor records'.